Source/Laser/LaserProfilesImpl/LaserProfileFromFile.cpp

```diff
@@ -0,0 +1,570 @@
+    amrex::Abort(Utils::TextMsg::Err("WarpX has to be compiled with the option openPMD=ON to read a lasy file"));
+    amrex::ignore_unused(ppl, params);
+#endif
+    } else if (lasy_file_name.empty() && !binary_file_name.empty()){
+        parse_binary_file(binary_file_name);
+    }
+    else if (!lasy_file_name.empty() && !binary_file_name.empty()){
+        Abort("Specify only one file format to load, either lasy_file_name or binary_file_name");
+    }
+    else{
+        Abort("binary_file_name or lasy_file_name must be provided for laser profile!");
+    }
+
+    //Set time_chunk_size
+    m_params.time_chunk_size = m_params.nt;
+    int temp = 1;
+    if(utils::parser::queryWithParser(ppl ,"time_chunk_size", temp)){
+        m_params.time_chunk_size = min(
+            temp, m_params.time_chunk_size);
+    }
+    if(m_params.time_chunk_size < 2){
+        Abort("Error! time_chunk_size must be >= 2!");
+    }
+    //Reads the (optional) delay
+    utils::parser::queryWithParser(ppl, "delay", m_params.t_delay);
+
+    //Allocate memory for E_lasy_data or E_binary_data Vector
+    const int data_size = m_params.time_chunk_size*
+            m_params.nx*m_params.ny;
+    if (m_params.file_in_lasy_format){
+        m_params.E_lasy_data.resize(data_size);
+    } else{
+        m_params.E_binary_data.resize(data_size);
+    }
+    //Read first time chunk
+    if (m_params.file_in_lasy_format){
+    read_data_t_chunk(0, m_params.time_chunk_size);
+    } else{
+    read_binary_data_t_chunk(0, m_params.time_chunk_size);
+    }
+    //Copy common params
+    m_common_params = params;
+}
+
+void
+WarpXLaserProfiles::FromFileLaserProfile::update (amrex::Real t)
+{
+
+    t += m_params.t_min - m_params.t_delay;
+    if(t >= m_params.t_max)
+        return;
+    const auto idx_times = find_left_right_time_indices(t);
+    const auto idx_t_left = idx_times.first;
+    const auto idx_t_right = idx_times.second;
+    //Load data chunk if needed
+    if(idx_t_right >  m_params.last_time_index){
+        if (m_params.file_in_lasy_format){
+        read_data_t_chunk(idx_t_left, idx_t_left+m_params.time_chunk_size);
+        } else{
+        read_binary_data_t_chunk(idx_t_left, idx_t_left+m_params.time_chunk_size);
+        }
+    }
+}
+
+void
+WarpXLaserProfiles::FromFileLaserProfile::fill_amplitude (
+    const int np,
+    Real const * AMREX_RESTRICT const Xp, Real const * AMREX_RESTRICT const Yp,
+    Real t, Real * AMREX_RESTRICT const amplitude) const
+{
+    t += m_params.t_min - m_params.t_delay;
+    //Amplitude is 0 if time is out of range
+    if(t < m_params.t_min ||  t > m_params.t_max){
+        amrex::ParallelFor(np,
+            [=] AMREX_GPU_DEVICE (int i) {
+                amplitude[i] = 0.0_rt;});
+        return;
+    }
+    //Find left time index
+    const auto idx_t_left = find_left_right_time_indices(t).first;
+    if(idx_t_left <  m_params.first_time_index){
+        Abort("Something bad has happened with the simulation time");
+    }
```

Comment:
I would suggest this change
```suggestion
    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(idx_t_left >=  m_params.first_time_index),
        "Something bad has happened with the simulation time");
```